compiler-backend/javascript/src/convert/generator/functional/render.rs

```
@@ -338,6 +403,380 @@ fn render_source_functions(renderer: &mut ModuleRenderer<'_, '_, '_, '_>) -> Mod
+        }
+        let Some((first, remaining)) = arguments.split_first() else {
+            unreachable!("invariant violated: curried tail-call function has no parameters")
+        };
+        writer.function(name, vec![first.clone()], exported, |writer| {
+            self.render_curried_tail_call_wrapper(
+                tree, writer, group, profile, state, &arguments, remaining, context,
+            )
+        })
+    }
+
+    fn render_local_tail_call_wrapper(
```

Clippy (1.97.1)

[warning] this function has too many arguments (8/7)

(warning)